alchemy/src/cloudflare/ai-gateway.ts

```diff
@@ -0,0 +1,301 @@
+ * });
+ */
+export const AiGateway = Resource(
+  "cloudflare::AiGateway",
+  { alwaysUpdate: true }, // AI Gateway PUT updates, so always run update logic
+  async function (
+    this: Context<AiGateway>,
+    id: string,
+    props: AiGatewayProps
+  ): Promise<AiGateway> {
+    const api = await createCloudflareApi(props);
+    const gatewayPath = `/accounts/${api.accountId}/ai-gateway/gateways/${id}`;
+    const gatewaysPath = `/accounts/${api.accountId}/ai-gateway/gateways`;
+
+    if (this.phase === "delete") {
+      try {
+        const deleteResponse = await api.delete(gatewayPath);
+        if (!deleteResponse.ok && deleteResponse.status !== 404) {
+          await handleApiError(deleteResponse, "delete", "ai gateway", id);
+        }
+      } catch (error) {
+        // Log error but don't throw, allow destroy to proceed
+        console.error(`Error deleting AI Gateway ${id}:`, error);
+      }
+      return this.destroy();
+    }
+
+    // Default values as per Cloudflare documentation examples
+    const mergedProps: AiGatewayProps = {
+      cacheInvalidateOnUpdate: true,
+      cacheTtl: 0,
+      collectLogs: true,
+      rateLimitingInterval: 0,
+      rateLimitingLimit: 0,
+      rateLimitingTechnique: "fixed",
+      ...props, // User props override defaults
+    };
+
+    let response: Response;
+    let apiResource: any;
+
+    try {
+      if (this.phase === "update") {
+        // Update existing gateway (PUT request)
+        const requestBody = mapPropsToApi(id, mergedProps, false);
+        response = await api.put(gatewayPath, requestBody);
+      } else {
+        // Check if gateway exists before creating (to avoid conflict)
+        const getResponse = await api.get(gatewayPath);
+        if (getResponse.status === 200) {
+          // Gateway exists, treat as update (PUT)
+          console.log(
+            `AI Gateway '${id}' already exists. Updating existing resource.`
+          );
+          const requestBody = mapPropsToApi(id, mergedProps, false);
+          response = await api.put(gatewayPath, requestBody);
+        } else if (getResponse.status === 404) {
+          // Gateway doesn't exist, create new (POST request)
+          const requestBody = mapPropsToApi(id, mergedProps, true); // Include 'id' in POST body
+          response = await api.post(gatewaysPath, requestBody);
+        } else {
+          // Unexpected error during GET check
+          await handleApiError(getResponse, "get", "ai gateway", id);
+          throw new Error("Unexpected error checking for existing gateway"); // Should not happen if handleApiError throws
+        }
+      }
+
+      if (!response.ok) {
+        const action = this.phase === "update" ? "update" : "create";
+        await handleApiError(response, action, "ai gateway", id);
+        // handleApiError should throw, but satisfy TS compiler
+        throw new Error(`Failed to ${action} AI Gateway ${id}`);
```

Comment:
`hanldeApiError` throws for you. It's a weird function that needs to be async to consume the `text()` or `json()` stream. You don't need the `throw` below